### ISSUE TITLE
prediction

### ISSUE THREAD

**Issue by Meng-J** [erprogs/CViT#18]
For the model I trained myself, the highest accuracy of the verification set is 88%, and the predicted value of the test set is only about 50%, which is different from the loss value of 91.5% and 0.32 in your paper. Could you please help me find out what the problem is? My data amount is about 270 million pictures, looking forward to your reply
![预测结果截图10-10-43](https://github.com/erprogs/CViT/assets/54169931/672576de-9360-4a66-b4ce-12914d00755d)

**Comment by erprogs** [comment]
Hello @Meng-J,

For 270 million pictures, the model's accuracy should definitely be higher. Could you share the learning curve? you can find it in weight folder.

We've updated our model—please update your repo to use the latest version. Check the README file for more information.

The current model is trained on 5 datasets and it's performing really well, both during training and prediction. Using your dataset, I expect you'll have an excellent deepfake predictor.

**Comment by Meng-J** [comment]
![50原模型](https://github.com/erprogs/CViT/assets/54169931/0f42ebc1-c588-46b9-98b4-edb237c5d118)
![50原模型acc](https://github.com/erprogs/CViT/assets/54169931/ab08ae55-1650-4128-a37d-547c2213ea34)
These two are the results of training on my own computer using your training code
![Figure_1](https://github.com/erprogs/CViT/assets/54169931/badae435-6352-4c1b-bca6-0d97b23c8b9a)
This is the predicted value of the model trained by you, and the result is indeed the same as that of the paper. However, the model I trained by myself has a poor effect and can only reach the prediction accuracy of about 50%. I did not make any other changes, except that the amount of data is much larger than yours, but I also used your method for data preprocessing, so I am very confused about my result. Could you take a look at it for me?

**Comment by erprogs** [comment]
The model is overfitting and might be overwhelmed by the very large data. For contrast, you can look at our CViT2 learning curve, which is trained on 1 million images with 90% augmentation ~ 1.9mil images.

Additionally, I have fixed a design error that prevented the model to learn in its full capacity.

I have changed 

    `self.pos_embedding = nn.Parameter(torch.randn(32, 1, dim))`
to 
  `self.pos_embedding = nn.Parameter(torch.randn(1, self.max_sequence_length, dim))`

then 
  `x += self.pos_embedding[0:shape]`
to 
  `x += self.pos_embedding[:, :x.size(1)]`  

Could you train your model again on the new training code?

![image](https://github.com/erprogs/CViT/assets/28438821/3fe3fc9d-768d-4aa9-b275-4823e11df7f5)

**Comment by Meng-J** [comment]
I don't see your new model

**Comment by Meng-J** [comment]
I don't see your new model

**Comment by erprogs** [comment]
The model is in this repository, so you will need to update your local clone. The trained weights are stored on [Hugging Face.](https://huggingface.co/datasets/Deressa/cvit) 
The details are in the README file.
